### ISSUE TITLE
Hidden fields being corrupted on save [possible bug]

### ISSUE THREAD

**Issue by nicogauthierpin** [Hube2/acf-user-role-field-setting#21]
I noticed this issue today where some hidden fields were corrupted when saving the post. These fields are text fields that included double quotes in them. For example:

`<rect x="13.2695" y="1397.12732" width="101.28678" height="140.37125"/>`

This was cropped to `<rect x="` when a user that had this field hidden saved the post. In addition, they'd see this:

<img width="1111" alt="Screen Shot 2020-12-04 at 6 06 28 PM" src="https://user-images.githubusercontent.com/10453794/101231583-15c6a680-3661-11eb-9109-d469a743dcf0.png">

Looks like double quotes within a hidden field are not being escaped correctly, or something along those lines.

This is on WP version 5.5.3, ACF Pro version 5.9.3 , ACF User Role Field Setting 3.0.2.

**Comment by Hube2** [comment]
Can you tell me what the value of the hidden field is? does in contain quote (") character?

**Comment by Hube2** [comment]
After some testing it appears this is because the field value contains HTML. 

I cannot fix this and preserve HTML entered into field content in this plugin.

What you will need to do is create an acf/prepare_field filter for this field. https://www.advancedcustomfields.com/resources/acf-prepare_field/. In this filter you will need to use htmlentities() https://www.php.net/manual/en/function.htmlentities.php to change the html tags to entity characters. You will need to do this based on if the field is hidden or not. See the method "user_can_edit" https://github.com/Hube2/acf-user-role-field-setting/blob/master/acf-user-role-field-setting.php#L45

Then you will need to create an acf/update_value filter https://www.advancedcustomfields.com/resources/acf-update_value/ and in this filter, again based on if the user can edit the field or not, undo what was done in the first filter by using html_entity_decode https://www.php.net/manual/en/function.html-entity-decode.php

**Comment by Hube2** [comment]
This is a problem that I had not considered and I will continue to look at a possible solution, but for now I suggest that if you need to use HTML in a field that is disabled for some users that you build your own work-a-round as explained above.

**Comment by Rasmusbdk** [comment]
I'm probably missing something obvious here, but why are you even outputting the hidden fields in the first place? Isn't ACF designed to only update/save the fields that are present in the post query? :-)

> This is a problem that I had not considered and I will continue to look at a possible solution, but for now I suggest that if you need to use HTML in a field that is disabled for some users that you build your own work-a-round as explained above.

**Comment by Hube2** [comment]
The reason that I started outputting hidden field was do to trying to deal with repeaters being reordered. In order to make sure the repeaters and flex fields were updated properly the contents of those fields needed to be submitted. If not submitted they would still be associated with the old row index.

Overall, this was a bad idea. It was a bad idea to try and include repeater fields in any way. Actually it did not. but then I got requests to support sub fields and here we are. I should have thought it out more before I added it. There is simply no way to deal to easily deal with the issue of reordering rows in a repeater.

Quite honestly, this should not be a plugin and should be dealt with by individual developers by adding their own acf/prepare_field filters. I created it when I was new to ACF (before this filter even existed) and needed the functionality and thought some others would like it as well before I knew about the issues caused by repeaters. Anyone that is doing development that needs functionality of this nature should be able to create the code needed by their specific requirement. If help is needed then there is the ACF help forum.
